File: src/Chapter-5-C99/Questions/Question2.c
//Write a program that asks the user for a 24-hour time, then displays the time in
//12-hour form:
//Enter a 24-hour time: 21:11
//Equivalent 12-hour time: 9:11PM
#include <stdio.h>

int main(void)
{
    int hour, minute;
    printf("Enter a 24-hour time: ");
    scanf("%d:%d", &hour, &minute);

    //if hour less than or equal to 12, then
    //its all good
    if (hour <= 12)
    {
        printf("Equivalent 12-hour time: %d:%d AM\n", hour, minute);
    }
    else
    {
        printf("Equivalent 12-hour time: %d:%d PM\n", hour-12, minute);
    }

    return 0;
}

File: src/Chapter-5-Rust/Questions/Question2.rs
//Write a program that asks the user for a 24-hour time, then displays the time in
//12-hour form:
//Enter a 24-hour time: 21:11
//Equivalent 12-hour time: 9:11PM
#[macro_use] extern crate text_io;
//process library to pause console
use std::process::Command;
//Need this to use the print!
use std::io::{self, Write};

fn main()
{
	let (hour, minute) : (i32, i32);
	print!("Enter a 24-hour time: ");
	io::stdout().flush().unwrap();
	//Need a space between for it to work
	scan!("{} : {}", hour, minute);
	//if hour less than or equal to 12, then
    //its all good
    if hour <= 12
    {
    	print!("Equivalent 12-hour time: {}:{} AM\n", hour, minute);
    }
    else
    {
    	print!("Equivalent 12-hour time: {}:{} PM\n", hour-12, minute);
    }
    io::stdout().flush().unwrap();
	let _ = Command::new("cmd.exe").arg("/c").arg("pause").status();
}
